pkg/collector/corechecks/containers/containerd/check.go

```diff
@@ -140,6 +155,58 @@ func (c *ContainerdCheck) runContainerdCustom(sender sender.Sender) error {
 	return nil
 }
 
+func toSnakeCase(s string) string {
+	snake := matchAllCap.ReplaceAllString(s, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func (c *ContainerdCheck) collectImagePullMetrics(sender sender.Sender) error {
+	openmetricsEndpoint := fmt.Sprintf("%s/v1/metrics", c.instance.OpenmetricsEndpoint)
+	resp, err := c.httpClient.Get(openmetricsEndpoint)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	parsedMetrics, err := prometheus.ParseMetrics(body)
+	if err != nil {
+		return err
+	}
+
+	for _, sample := range parsedMetrics {
```

Comment:
I think we should define a list of metrics that we support and define transformers for them. Otherwise we'll be sensitive to any change in containerd.

Reply:
In this specific case, we are not getting all metrics. 

We are only cherry picking the image pull metric. 
All other metrics are skipped here.

I think it is an overkill to define a list of supported metrics and transformers only for one metric only (which is `containerd.image.pull` ). Even if we define a list of metrics we support (which is only 1 metric in this case) and define transformers for them, then if containerd changes the way it exposes the metrics, we would still need to modify the implementation depending on what changes with containerd. And in any case, the existing implementation would not break the check, it would simply stop reporting the metric.

What do you think?

Reply:
The advantage of using transformers are:
- Easily extensible
- Follows the same pattern as other checks
- We won't introduce other metrics by mistake

Indeed we'll need to follow containerd's changes and adapt to them but I think it's worth it. What do you think ?

Reply:
indeed we need a `include metrics` list to drop the unexpected metrics

Reply:
Makes sense

Thank you for the suggestion

I refactored the code to use transformers to include the metrics that we need to collect and ignore other metrics.